WordPress/Classes/Services/AccountService.m

```diff
@@ -406,29 +405,35 @@ - (void)updateAccount:(WPAccount *)account withUserDetails:(RemoteUser *)userDet
     account.displayName = userDetails.displayName;
     account.dateCreated = userDetails.dateCreated;
     account.emailVerified = @(userDetails.emailVerified);
-    if (userDetails.primaryBlogID) {
-        [self configurePrimaryBlogWithID:userDetails.primaryBlogID account:account];
-    }
+    account.primaryBlogID = userDetails.primaryBlogID;
+
+    [self updateDefaultBlogIfNeeded: account];
```

Comment:
TIL!  Thanks for that tip. My first take on this was to rely on `willSave` to ensure the the defaultBlog was updated.  A bit overkill to do it that way.  This would be much nicer. :)

Reply:
Side note... It doesn't seem proper for a model to call out to a service to work, which is partly why I ditched the `willSave` approach--the service seemed best suited for updating the extensions.  That would be my push-back on doing the work in the model's setter.  
Another point if favor of nuking the relation? 😁

Reply:
Yeah, I think I'd agree with that – maybe it doesn't make sense for the model to do this.